apps/extension/src/state/wallets.ts

```diff
@@ -35,6 +38,30 @@ export const createWalletsSlice =
         await local.set('wallets', [newWallet.toJson(), ...wallets]);
         return newWallet;
       },
+      getSeedPhrase: async () => {
+        const passwordKey = get().password.key;
+        const key = await Key.fromJson(passwordKey!);
+
+        const activeWallet = getActiveWallet(get());
+
+        if (!activeWallet) return [];
+
+        const encryptedSeedPhrase = activeWallet.custody.encryptedSeedPhrase;
+        const unsealeSeedPhrase = await key.unseal(encryptedSeedPhrase);
+
+        return unsealeSeedPhrase?.split(' ') ?? [];
```

Comment:
Making `[]` to mean `no seed phrase` is not the best approach. We should either make it return `string[] | undefined` or just `string[]` and throw errors if someone attempts to call this without the necessary requirements. This is what it would look like:

```typescript
      getSeedPhrase: async () => {
        const passwordKey = get().password.key;
        if (!passwordKey) throw new Error('no password set');

        const key = await Key.fromJson(passwordKey);
        const activeWallet = getActiveWallet(get());
        if (!activeWallet) throw new Error('no wallet set');

        const decryptedSeedPhrase = await key.unseal(activeWallet.custody.encryptedSeedPhrase);
        if (!decryptedSeedPhrase) throw new Error('Unable to decrypt seed phrase with password');

        return decryptedSeedPhrase.split(' ');
      },
```